File: frontend/src/types/LabInstance/LabInstanceTask.ts
// src/types/LabInstance/LabInstanceTask.ts
// =============================================================================
// LAB INSTANCE TASK TYPES (matches backend schemas)
// =============================================================================

export interface LabInstanceTask {
    id: string
    lab_instance_id: string
    task_type: string
    status: string
    enqueued_at: string | null
    started_at: string | null
    finished_at: string | null
    worker_pid: number | null
    worker_host: string | null
    error_message: string | null
    created_at: string | null
}

export interface LabInstanceTaskList {
    items: LabInstanceTask[]
    total: number
}

File: backend/app/schemas/LabDefinition/LabInstanceTask.py
# backend/app/schemas/LabDefinition/LabInstanceTask.py
from datetime import datetime
from typing import Optional
from uuid import UUID
from pydantic import BaseModel


class LabInstanceTaskResponse(BaseModel):
    id: UUID
    lab_instance_id: UUID
    task_type: str
    stage: Optional[str] = None
    status: str
    progress_percent: Optional[int] = None
    enqueued_at: Optional[datetime] = None
    started_at: Optional[datetime] = None
    finished_at: Optional[datetime] = None
    worker_pid: Optional[int] = None
    worker_host: Optional[str] = None
    error_message: Optional[str] = None
    created_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class LabInstanceTaskListResponse(BaseModel):
    items: list[LabInstanceTaskResponse]
    total: int